src/m3ninx/index/segment/fst/segment.go

```diff
@@ -922,6 +958,32 @@ func (sr *fsSegmentReader) Docs(pl postings.List) (doc.Iterator, error) {
 	return iter, err
 }
 
+func (sr *fsSegmentReader) EncodedDoc(id postings.ID) (doc.EncodedDocument, error) {
+	if sr.closed {
+		return doc.EncodedDocument{}, errReaderClosed
+	}
+	// NB(r): We are allowed to call match field after Close called on
+	// the segment but not after it is finalized.
+	sr.fsSegment.RLock()
+	pl, err := sr.fsSegment.encodedDocNotClosedMaybeFinalizedWithRLock(id)
```

Comment:
We should pass an encoded doc reader that is allocated when the `fsSegmentReader` is created in here so that the reader has it's own encoded doc reader that is not concurrently accessed.

Reply:
Ahh, my mistake. I misread the function signature as `Lock` and not `RLock`. Passing in from the segment reader makes sense. Will fix this up.

Reply:
Ah not to mind, thought this expanded to a doc.Document. Makes sense.